networkx/algorithms/similarity.py

```diff
@@ -1550,6 +1583,18 @@ def panther_similarity(
     """
     import numpy as np
 
+    if source not in G:
+        raise nx.NodeNotFound(f"Source node {source} not in G")
+
+    isolates = list(nx.isolates(G))
```

Comment:
I think you only use the isolates by checking membership.
That is `O(len(list))` for a list, but O(1) for a set.
I think you don't need them to be ordered. So maybe its better to use a set instead of a list.